supervision/metrics/core.py

```diff
@@ -0,0 +1,210 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Any, Dict, Iterator, Optional, Tuple, Union
+
+import numpy as np
+import numpy.typing as npt
+
+from supervision import config
+from supervision.detection.core import Detections
+
+"""Used by metrics module as class ID, when none is present"""
+CLASS_ID_NONE = -1
+
+
+class Metric(ABC):
+    """
+    The base class for all supervision metrics.
+    """
+
+    @abstractmethod
+    def update(self, *args, **kwargs) -> Metric:
+        """
+        Add data to the metric, without computing the result.
+        Return the metric itself to allow method chaining, for example:
+
+        Example:
```

Comment:
I think we can skip the examples in base class.

Reply:
removed